1.7.0 SDK Release (#51)
klaviyo/klaviyo-swift-sdk: README.md

KlaviyoSwift is available via [Swift Package Manager (SPM)](https://swift.org/package-manager/). Follow the steps below to get it setup.

### Import the SDK

> Open your project and navigate to your project’s settings. Select the Swift Packages tab and click on the add button below the packages list. Enter the URL of our Swift SDK repository (https://github.com/klaviyo/klaviyo-swift-sdk) in the text field and click Next. On the next screen, select the SDK version (1.7.0 as of this writing) and click Next.

Reviewer: Would it be easier to comprehend this if it were a list vs. a paragraph?

Author: I think for now might leave it. I'd like to do a more major revamp on the README as a whole.

## Anonymous Tracking Notice

> By default, Klaviyo will begin tracking unidentified users in your app once the SDK is initialized. This means you will be able to track events from users in your app without any user information provided. When an email or other primary identifier is provided Klaviyo will merge the data from the anonymous user to a new identified user. Prior to version 1.7.0, the Klaviyo SDK used the [IDFV](https://developer.apple.com/documentation/uikit/uidevice/1620059-identifierforvendor) to facilitate anonymous tracking. After that version the SDK will use a cached UUID that is generated when the SDK is initalized. The SDK will also detect if an IDFV was used previously as an identifier and will use that instead of generating a new UUID.

Reviewer 2: I might tweak this section slightly. Changes in **bold**

Prior to version 1.7.0, the Klaviyo SDK used the **Apple identifier for vendor (IDFV)** (https://developer.apple.com/documentation/uikit/uidevice/1620059-identifierforvendor) to facilitate anonymous tracking. **Starting with version 1.7.0,** the SDK will use a cached UUID that is generated when the SDK is **initialized**. **For existing anonymous profiles using IDFV, the SDK will continue to use IDFV, instead of generating a new UUID.**

Author: thx!

Author: ```suggestion
By default, Klaviyo will begin tracking unidentified users in your app once the SDK is initialized. This means you will be able to track events from users in your app without any user information provided. When an email or other primary identifier is provided Klaviyo will merge the data from the anonymous user to a new identified user. Prior to version 1.7.0, the Klaviyo SDK used the [Apple identifier for vendor (IDFV)](https://developer.apple.com/documentation/uikit/uidevice/1620059-identifierforvendor) to facilitate anonymous tracking. Starting with version 1.7.0, the SDK will use a cached UUID that is generated when the SDK is initialized. For existing anonymous profiles using IDFV, the SDK will continue to use IDFV, instead of generating a new UUID.
```

If your user taps on the notification this will be tracked back to Klaviyo as an "Opened Push" event assuming you have implemented the tracking changes discussed above.

## SDK Data Transfer

> As version 1.7.0 the SDK will cache incoming data and flush it back to the Klaviyo API on an interval. As of this writing the interval is based on the network link currently being used by the app. The table below shows the flush interval used for each type of connection:

Reviewer 2: I'd change "As version 1.7.0" to "Starting with version 1.7.0"

Author: ```suggestion
Starting with version 1.7.0, the SDK will cache incoming data and flush it back to the Klaviyo API on an interval. As of this writing the interval is based on the network link currently being used by the app. The table below shows the flush interval used for each type of connection:
```

> As version 1.7.0 the Klaviyo will cache incoming data and flush it back to the Klaviyo API on an interval. As of this writing the interval is based on the network link currently being used by the app. The table below shows the flush interval used for each type of connection:

Reviewer: ```suggestion
As version 1.7.0 Klaviyo will cache incoming data and flush it back to the Klaviyo API on an interval. As of this writing the interval is based on the network link currently being used by the app. The table below shows the flush interval used for each type of connection:
```

Author: I think I meant SDK here....

Author: ```suggestion
As version 1.7.0 the SDK will cache incoming data and flush it back to the Klaviyo API on an interval. As of this writing the interval is based on the network link currently being used by the app. The table below shows the flush interval used for each type of connection:
```

| Network     | Interval    |
| :---        | :--- |
| WWAN/Wifi   | 10 seconds  |
| Cellular    | 30 seconds  |

Connection determination is based on notifications from our reachability service. When there is no network available the SDK will cache data until the network becomes available again. All data sent by the SDK should be available shortly after it is flushed by the SDK. 

### Retries

> The SDK will retry API requests that fail under conditions. For example if a network timeout occurs the request will be retried on the next flush inteval. In addition if the SDK received a 429 from the Klaviyo API it will use expontential backoff with jitter to retry the next request.

Reviewer 2: Should it be "fail under certain conditions"?

Also even though most people will know what a 429 error is, I might specify that it's a rate limiting error... In addition, if the SDK received a 429 rate limiting error from the Klaviyo API, it will use exponential backoff with jitter to retry the next request.

Author: ```suggestion
The SDK will retry API requests that fail under certain conditions. For example if a network timeout occurs the request will be retried on the next flush interval. In addition if the SDK receives a rate limiting error (429) from the Klaviyo API it will use exponential backoff with jitter to retry the next request.
```
